internal/security/validator.go

```diff
@@ -131,28 +131,45 @@ func (v *Validator) validateAccessLevel(command string, readOperations []string)
 
 // isReadOperation checks if a command is a read operation
 func (v *Validator) isReadOperation(command string, allowedOperations []string) bool {
+	// Check if the command contains help flags - these are always read-only
+	if strings.Contains(command, "--help") || strings.Contains(command, " -h ") || strings.HasSuffix(command, " -h") {
+		return true
+	}
+
 	// Normalize command by removing any options/arguments
 	// This extracts the base command like "az aks show" from "az aks show --name myCluster"
 	cmdParts := strings.Fields(command)
-	var baseCommand string
-
-	if len(cmdParts) >= 3 && cmdParts[0] == CommandTypeAz {
-		// Take the first three parts (e.g., "az aks show")
-		baseCommand = strings.Join(cmdParts[:3], " ")
-	} else if len(cmdParts) == 2 && cmdParts[0] == CommandTypeAz {
-		// Handle commands with just two parts (e.g., "az version")
-		baseCommand = strings.Join(cmdParts[:2], " ")
+	
+	if len(cmdParts) == 0 || cmdParts[0] != CommandTypeAz {
+		return false
 	}
 
-	// Now check if the base command is in our list of allowed read operations
+	// For az commands, we need to handle various command structures:
+	// - "az version" (2 parts)
+	// - "az aks show" (3 parts)
+	// - "az aks check-network outbound" (4 parts)
+	// - "az aks trustedaccess rolebinding list" (5 parts)
+	// - "az aks nodepool get-upgrades" (4 parts)
```

Comment:
how about command like 'az aks get | grep id'? As it may be any random commands after |, we should reject such shell commands directly